core/src/test/java/org/mskcc/oncotree/utils/VersionUtilTest.java

```diff
@@ -0,0 +1,271 @@
+                    ") and expected version list (size:" + Integer.toString(expectedVersionMap.size()) + ")");
+            failureCount = failureCount + 1;
+        }
+        if (failureCount > 0) {
+            fail(Integer.toString(failureCount) + " failed test conditions. Details follow... " + failureReport.toString());
+        }
+        return;
+    }
+
+    @Test
+    public void testSuccessfulVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = expectedVersionMap.keySet().iterator().next();
+        Version expectedVersion = expectedVersionMap.get(testVersionName);
+        Version returnedVersion = versionUtil.getVersion(testVersionName);
+        int failureCount = 0;
+        StringBuilder failureReport  = new StringBuilder();
+        if (expectedVersion == null || returnedVersion == null) {
+            if (expectedVersion == null) {
+                failureReport.append(testVersionName + " : no values in expected version hash (internal error)\n");
+            }
+            if (returnedVersion == null) {
+                failureReport.append(testVersionName + " : version expected but not returned from versionUtil\n");
+            }
+        } else {
+            String expectedVersionName = expectedVersion.getVersion();
+            String returnedVersionName = returnedVersion.getVersion();
+            if (!testValuesMatch(expectedVersionName, returnedVersionName)) {
+                failureReport.append(makeMismatchMessage(testVersionName, "version", returnedVersionName, expectedVersionName));
+                failureCount = failureCount + 1;
+            }
+            String expectedGraphURI = expectedVersion.getGraphURI();
+            String returnedGraphURI = returnedVersion.getGraphURI();
+            if (!testValuesMatch(expectedGraphURI, returnedGraphURI)) {
+                failureReport.append(makeMismatchMessage(testVersionName, "graphURI", returnedGraphURI, expectedGraphURI));
+                failureCount = failureCount + 1;
+            }
+            String expectedDescription = expectedVersion.getDescription();
+            String returnedDescription = returnedVersion.getDescription();
+            if (!testValuesMatch(expectedDescription, returnedDescription)) {
+                failureReport.append(makeMismatchMessage(testVersionName, "description", returnedDescription, expectedDescription));
+                failureCount = failureCount + 1;
+            }
+        }
+        if (failureCount > 0) {
+            fail(Integer.toString(failureCount) + " failed test conditions. Details follow... " + failureReport.toString());
+        }
+        return;
+    }
+
+    @Test(expected = InvalidVersionException.class)
+    public void testUnmatchedVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = "";
+        for (String versionName : expectedVersionMap.keySet()) {
```

Comment:
So this is just appending all the versions together into one long string? Is this better than just hardcoding some value you know will never be a version name?